[v1 migration guide] Grammar and vocabulary improvements (#692)
pedronauck/docz: MIGRATION_GUIDE.md

# Migration Guide

> The [v1 release](https://github.com/pedronauck/docz/pull/656) was one of our big releases so far and a lot of breaking changes was introduced. So, there's a few apis that changed and you need to update your code if you're coming from previous versions. It's not a big deal, but you need to follow this guide in order to get Docz running properly on your project after an upgrade.

Reviewer: ```suggestion
The [v1 release](https://github.com/pedronauck/docz/pull/656) was one of our big releases so far and a lot of breaking changes were introduced. A few APIs were changed and you will need to update your code if you're coming from a previous version. It's not a big deal, but you need to follow this guide in order to get Docz running properly on your project after the upgrade.
```

## Update React to use Hooks

We made a [huge improvement](https://github.com/pedronauck/docz/commit/f57f987df0536b3b65a26f1b0e8a8f8f00d63800) on docz using the new react hooks. So, the biggest requirement is that you need `react` and `react-dom` with the version `>= 16.8.0`, because that's the version that has hooks released and stable. So, just update your React version, it's fully retro compatible.

## Spectrum instead of Discord

> Another thing that we changed, it's now we're attending on Spectrum instead of Discord. You can check the [Docz community](https://spectrum.chat/docz) and ask us whatever you want!

Reviewer: ```suggestion
Another thing that has changed is our community is now on Spectrum instead of Discord. You can check the [Docz community](https://spectrum.chat/docz) and ask us whatever you want!
```

## No more render props

> In the older version of docz, we're using render props as data components in order to get data from the docz database and use it on themes. Now, all this render props became hooks. This is a huge improvement, since it's much easier to use them.

Reviewer: Should we capitalise `Docz` here to remain consistent? :+1:

## Order deprecated removed

> Since [v0.12.4](https://github.com/pedronauck/docz/releases/tag/v0.12.4) we launched `menu` property to create and sort the menu, and the `ordering` frontmatter field was deprecated. So, now we're removing this property. If you wanna see more information about the `menu` order property, you can take a look at the `Ordering` session on our website.

Reviewer: ```suggestion
In [v0.12.4](https://github.com/pedronauck/docz/releases/tag/v0.12.4) we launched the `menu` property to create and sort menu items. At the same time the `ordering` frontmatter field was deprecated. We are now going to remove this deprecated field. If you want to see more information about the `menu` order property, you can take a look at the `Ordering` section on the Docz website.
```

## Remove hash router support

> In the newest version of docz, because of some performance and bundle issues, we are now using `@reach/router` instead of `react-router`. As `@reach/router` doesn't have an official support for hash routing yet, and as there are lots of good free services to host static sites besides Github pages ([Surge](https://surge.sh/), for instance, is free and has full support for browser history navigation) we decided to deprecate the hash router support.

Reviewer: ```suggestion
In the newest version of Docz, because of some performance and bundle issues, we are now using `@reach/router` instead of `react-router`. As `@reach/router` doesn't have official support for hash routing yet, and as there are lots of good free services to host static sites besides Github pages ([Surge](https://surge.sh/), for instance, is free and has full support for browser history navigation) we have decided to deprecate hash router support.
```

## Creating and using Docz themes

The process to create themes for docz is very similar to the previous one; there are no big changes here, but you need to know a few changes that we made.

- The first one, is you don't have `DocPreview` anymore. Instead, we introduced the `ComponentsProvider` component.

> - The second one, is that the `render` field previously passed in the components mapper now is called `playground`.

Reviewer: ```suggestion
- The second one is that the `render` field previously passed in the components mapper; now it is called `playground`.
```
